Modules/CPackDeb.cmake

```diff
@@ -674,18 +704,26 @@ if(CPACK_DEB_PACKAGE_COMPONENT)
   get_component_package_name(CPACK_DEBIAN_PACKAGE_NAME "${CPACK_DEB_PACKAGE_COMPONENT}")
 endif()
 
+if (NOT CPACK_DEBIAN_PACKAGE_GENERATE_SHLIBS_POLICY)
+  set(CPACK_DEBIAN_PACKAGE_GENERATE_SHLIBS_POLICY "=")
+endif()
+
 # Generate shlibs file
-foreach(_FILE ${CPACK_DEB_SHARED_OBJECT_FILES})
-  get_filename_component(_LIBNAME ${_FILE} NAME_WE)
-  get_filename_component(_LIBEXT ${_FILE} EXT)
+if(NOT DEFINED CPACK_DEBIAN_PACKAGE_GENERATE_SHLIBS)
+  set(CPACK_DEBIAN_PACKAGE_GENERATE_SHLIBS 1)
+endif()
```

Comment:
I think we should fail if user want to generate shlibs, but there is no readelf program